sp/src/vscript/vscript_squirrel.cpp

```diff
@@ -1555,29 +1552,27 @@ SQInteger constructor_stub(HSQUIRRELVM vm)
 		return sqstd_throwerrorf(vm, "Unable to construct instances of %s", pClassDesc->m_pszScriptName);
 	}
 
-	SquirrelVM* pSquirrelVM = (SquirrelVM*)sq_getsharedforeignptr(vm);
-	Assert(pSquirrelVM);
-
-	sq_resetobject(&pSquirrelVM->lastError_);
+	SQUserPointer p;
+	if (SQ_FAILED(sq_getinstanceup(vm, 1, &p, 0)))
 	{
-		SQUserPointer p;
-		if (SQ_FAILED(sq_getinstanceup(vm, 1, &p, 0)))
-		{
-			return SQ_ERROR;
-		}
+		return SQ_ERROR;
+	}
 
-		if (!p)
-		{
-			return sq_throwerror(vm, "Accessed null instance");
-		}
+	if (!p)
+	{
+		return sq_throwerror(vm, "Accessed null instance");
+	}
 
-		void* instance = pClassDesc->m_pfnConstruct();
+	void* instance = pClassDesc->m_pfnConstruct();
 
-		// expect construction to always succeed
-		Assert(sq_isnull(pSquirrelVM->lastError_));
+#ifdef DBGFLAG_ASSERT
+	SquirrelVM* pSquirrelVM = (SquirrelVM*)sq_getsharedforeignptr(vm);
+	Assert(pSquirrelVM);
+	// expect construction to always succeed
+	Assert(sq_isnull(pSquirrelVM->lastError_));
```

Comment:
Does this still need a `sq_resetobject(&pSquirrelVM->lastError_);` before `pClassDesc->m_pfnConstruct()`? I don't seem to have issues without it though. I guess `sq_release()` also nulls the object?